algosdk/logic.py

```diff
@@ -287,3 +289,82 @@ def get_application_address(appID: int) -> str:
     to_sign = constants.APPID_PREFIX + appID.to_bytes(8, "big")
     checksum = encoding.checksum(to_sign)
     return encoding.encode_address(checksum)
+
+
+class SourceMap:
```

Comment:
@barnjamin Optional:  I can imagine adding more sourcemap-related facilities later on.  For example, assuming PyTeal-to-TEAL sourcemaps exist, we may wish to provide facilities for chaining sourcemaps.

I raise the note because I'm wondering:  What do you think about relocating the functionality to a new file?

I don't feel _too_ strongly though I can imagine `logic.py` growing to be large over time.